pegasus/migrations/76_migrate_channel_sources_to_s3.rb

```diff
@@ -0,0 +1,60 @@
+require 'cdo/hip_chat'
+require 'files_api'
+
+# Migration for moving JSON.parse(row[:value])['levelSource'] and ['levelHtml'] to S3 (/v3/sources API).
+Sequel.migration do
+  up do
+    HipChat.log 'Moving <b>channel</b> sources to S3...'
+    FILE_NAME = 'main.json'
+
+    # For each channel with a level(Source|Html) and no migratedToS3 flag, create a new S3 record and set migratedToS3.
+    batch_update do |row|
+      begin
+        value = JSON.parse(row[:value])
+      rescue JSON::ParserError
+        next
```

Comment:
These unparsable channels are essentially "debt" that we have to work around in migrations.  I'd be in favor of running a one-time script to permanently delete these unparsable channels if they're over 30 days old.  We shouldn't be creating any new unparsable channels since they were a result of the [utf8mb4 charset issue](https://github.com/code-dot-org/code-dot-org/blob/staging/pegasus/migrations/73_modify_charset_on_storage_apps.rb) which was fixed on May 15, 2015.